packages/babel-parser/src/parser/expression.js

```diff
@@ -2389,12 +2391,22 @@ export default class ExpressionParser extends LValParser {
     checkKeywords: boolean,
     isBinding: boolean,
   ): void {
-    if (this.prodParam.hasYield && word === "yield") {
-      this.raise(startLoc, Errors.YieldBindingIdentifier);
+    // All JavaScript reserved words are not longer than 10 characters.
+    if (word.length > 10) {
+      return;
+    }
+    // Most identifiers are not reservedWord-like, they don't need special
+    // treatments afterward, which very likely ends up throwing errors
+    if (!canBeReservedWord(word)) {
```

Comment:
Or maybe we can move this check after the `if yield/await/arguments` checks, and just use `isStrictBindReservedWord`? Intuitively it shouldn't be slower, but I didn't check.